site/src/components/banner.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Banner
+description: The Banner component in the Atlas Design System
+template: standard
+figmaEmbed: https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2FuVA2amRR71yJZ0GS6RI6zL%2F%25F0%259F%258C%259E-Atlas-Design-Library%3Fnode-id%3D1002%253A7106%26t%3DWU64bU98uaT1nU7A-1
+classType: Component
+classPrefixes:
+  - banner
+---
+
+# Banner
+
+Banner component allows adding a full width notification block to the page.
+
+**This page is best viewed with the content section full screened because banners are intended to be full width elements. Go full screen by clicking the button in the top right corner.**
+
+## Usage
+
+Here is an example of a standard usage of banner component.
+
+```html-no-indent
+<section class="banner">
+	<p>A banner about something happening in the world <a href="#">Link Text</a></p>
+</section>
+```
+
+### Banner with title
+
+You can add a title to the banner with an optional icon.
+
+```html-no-indent
+<section class="banner">
+	<p class="banner-title">
+		<span class="icon" aria-hidden="true">
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 14"><path d="M7 10.3c-.6 0-1.2-.2-1.7-.5-.5-.2-.9-.6-1.2-1.1S3.7 7.6 3.7 7s.2-1.2.5-1.7.7-.9 1.2-1.2c1-.6 2.3-.6 3.4 0 .5.3.9.7 1.2 1.2s.5 1.1.5 1.7-.2 1.2-.5 1.7c-.4.5-.8.9-1.3 1.2-.5.3-1.1.4-1.7.4zm0-5.6c-.4 0-.8.1-1.1.3-.4.3-.6.5-.9.9-.2.3-.3.7-.3 1.1s.1.8.3 1.1c.3.4.5.6.9.9.7.4 1.6.4 2.2 0 .4-.3.6-.5.9-.9.2-.3.3-.7.3-1.1s-.1-.8-.3-1.1c-.3-.4-.5-.6-.9-.9-.3-.2-.7-.3-1.1-.3zM6.4 0h1.1v2.8H6.4zm0 11.2h1.1V14H6.4zm4.8-4.8H14v1.1h-2.8zm-1.561 3.606l.778-.778 1.98 1.98-.779.778zm.075-6.065l1.98-1.98.778.778-1.98 1.98zM0 6.4h2.8v1.1H0zm1.461-3.567l.778-.777 1.98 1.98-.778.777zm.067 8.429l1.98-1.981.777.777-1.978 1.98z"/></svg>
+        </span>
+		<span>Title</span>
+	</p>
+	<p>A banner about something happening in the world <a href="#">Link Text</a></p>
+</section>
+```
+
+### Dismissing banner
+
+Banners can be dismissed by adding the `data-dismissable` attribute to the `.banner` and `data-dismiss`to the closing button.
+
+If `data-dismissable` attribute is also assigned a `dissapearing` value - banner would slowly slide up.
+
+**Note**: banner's content should be wrapped up with `banner-content` container which helps with correct positioning and spacing of content.
+
+```html-no-indent
+<section
+	class="banner"
+	data-uid="example-banner-uid"
+	data-dismissable="disappearing"
+>
+	<div class="banner-content">
+		<button
+			type="button"
+			data-bi-name="close"
+            data-dismiss
+			data-banner-dismiss="example-banner-uid"
+			class="button-reset button-close line-height-1 border-radius-rounded position-absolute top-0 right-0"
```

Comment:
Back in docs-ui we used data-dismiss to indicate the button was a dismiss button and data-banner-dismiss to indicate the id of the banner being removed from the DOM. The banner had data-uid to indicate it's id.

Thinking we should replace data-uid with id and data-banner-dismiss with aria-owns for better a11y.

Question: is `section` the right element for a banner?

cc @BenDMyers @wibjorn

Reply:
Checking into the semantics; will report back

Reply:
**TL;DR:** `section` as it's used here is *probably* fine, but we *probably* want to consider encouraging the use of `aria-label` or `aria-labelledby`, *especially* for titled banners.

***

I wanted to see what accessibility practitioners were saying and doing about notification banners like these, so I checked on other design systems and component libraries that I knew were design-conscious, and I checked for prior conversations in an accessibility Slack I'm in.

## Slack Says Probably Not a Big Deal

The Slack conversations I saw didn't seem to have strong opinions on which roles were best for this, or whether a non-`generic` landmark role was even necessary for this. The mood seemed to be that a `div` (or an unlabeled `section` like this, which behaves identically to a `div`) could probably be fine, or even desirable in cases where we'd want to avoid overwhelming users with a clutter of landmarks. In some contexts, folks suggested using an `aside`, and this was also met with a "that's probably fine" shrug.

## GOV.UK Opts For `role=region`

The GOV.UK Design System, which is generally considered accessibility-conscious, uses `role=region` for [their Notification Banner component](https://design-system.service.gov.uk/components/notification-banner/). Checking through their GitHub issues, it seems like they've brought this through user testing with screenreader, and it tested positively, as users were able to identify the content and backtrack to it easily if they needed. GOV.UK is also using this pattern for cookie consent banners.

`role=region` (or a `section` that we want to inherit the `region` role) needs to have an accessible name defined by either `aria-label` or `aria-labelledby`. If we were to recommend the `role=region` approach, we'd probably also need to include a suggested `aria-label` (maybe a localized "Notice"?) or suggestions for how to use `aria-labelledby` (pointing to the title if one is provided) in this case.

## My Recommendation

This is *probably* fine to leave as is - an unlabeled `section` that'll take a `generic` role much like a `div`.

If we determined we needed stronger semantics, maintaining the `section` would be best, and the way to go would be to provide guidance around `aria-label`/`aria-labelledby`, using GOV.UK Design System as prior art.

## Elements and Roles That Don't Make the Cut

<details>
<summary>header and role=banner, weirdly enough</summary>
The header element supplies a "banner" role, which sounds perfect! However, the "banner" role is meant for <a href="https://w3c.github.io/aria/#banner">"A landmark that contains mostly site-oriented content, rather than page-specific content."</a> There's also not meant to be more than one "banner" role on a page.
</details>
<details>
<summary>role=alert</summary>
This creates a live region on the page, which means the screenreader may likely immediately start yammering these contents once the page loads. Also, the "alert" role isn't meant to have interactive controls like close buttons in them.
</details>

Reply:
Updated examples to use `div`'s with `id`'s and `aria-owns` on the closing button.

Also, after doing some investigation and giving it a thought, the example with the title was removed - it turned out they normally don't have standing-out titles.

Reply:
@jdanyow @BenDMyers @ollips I had to go and read up about `aria-owns`, and after doing so don't see why we should use it instead of maybe something like `aria-controls`.

> The only reason to include aria-owns is to expose a parent/child contextual relationship to assistive technology when the DOM's construction can't provide that relationship.

Quote from [aria-owns on mdn](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-owns).

So is a dismissible button the parent of the item it dismisses? I would say no. Compare to `aria-controls`:

> The global aria-controls property identifies the element (or elements) whose contents or presence are controlled by the element on which this attribute is set.

Reply:
Resolving this